pvanalytics/features/shading.py

```diff
@@ -0,0 +1,338 @@
+"""Functions for labeling shading and shodows."""
+import numpy as np
+import pandas as pd
+from scipy import ndimage
+from skimage import morphology, measure
+import pvlib
+from pvanalytics import util
+
+
+def _to_image(data, width):
+    """Convert data to an image.
+
+    Parameters
+    ----------
+    data : Series
+        Values of the pixels
+    width : int
+        Width of the image in pixels
+
+    Returns
+    -------
+    ndarray
+        Data reshaped into an ndarray with width `width`
+    """
+    return data.reshape(len(data) // width, width)
+
+
+def _to_series(image, image_index):
+    """Convert a binary image to a boolean series.
+
+    Parameters
+    ----------
+    image : ndarray
+        Binary image.
+    image_index : pd.Index
+        Index for the returned series.
+
+    Returns
+    -------
+    Series
+        A pandas Series with index `index` and values from the image.
+    """
+    return pd.Series(image.flatten(), index=image_index)
+
+
+def _smooth(image):
+    """Smooth `image` by local mean filtering.
+
+    Local means are computed in a 3x3 square surrounding each pixel.
+    """
+    return ndimage.uniform_filter(image, size=15)
```

Comment:
"3x3 square surrounding each pixel" like this?

X X X
X O X
X X X

I think `size=15` will be be a 15 x 15 square.  `size=(3, 3)`?

Maybe also `mode='nearest'` to handle windows extending outside the image.

Reply:
What was I doing?? Thanks. We could use `mode='nearest'`, but I think it wont make much of a difference since the smoothed image is just used to get a maximum ghi value to re-scale the input.

Reply:
> I think it wont make much of a difference since the smoothed image is just used to get a maximum ghi value to re-scale the input.

Ok with this, thanks.